packages/app/screens/landing/LandingScreen.js

```diff
@@ -1,67 +1,296 @@
-import React, { useCallback, useEffect } from 'react';
+import React, { useCallback, useRef } from 'react';
 import axios from 'axios';
-import { Text } from 'react-native';
+import { StyleSheet, View, Text, KeyboardAvoidingView, TouchableWithoutFeedback, Keyboard, Alert, TouchableOpacity, Platform} from 'react-native';
+
 import * as WebBroswer from 'expo-web-browser';
 import * as Google from 'expo-auth-session/providers/google';
 
 import Button from '@app/components/Button';
 import Screen from '@app/components/Screen';
+import TextInput from '@app/components/TextInput';
+import GoogleIcon from '@app/assets/google-icon';
 
 import { useUserContext } from '@app/utils/UserContext';
-import { getUserInfo } from '@app/utils/datalayer';
 import { ENDPOINT } from '@app/utils/constants';
+import { getUserInfo } from '@app/utils/datalayer';
+
+import EyeOff from '@app/assets/eye-line-off';
+import EyeOn from '@app/assets/eye-line-on';
 
 WebBroswer.maybeCompleteAuthSession();
 
 function LandingScreen() {
   const { user, setUser } = useUserContext();
 
-  const [request, response, promptAsync] = Google.useAuthRequest({
-    responseType: 'id_token',
-    expoClientId: '1080245081969-u9lnl9ospj757rq75kiumttqconhnfcc.apps.googleusercontent.com',
-    iosClientId: '1080245081969-u9lnl9ospj757rq75kiumttqconhnfcc.apps.googleusercontent.com',
-    androidClientId: '1080245081969-u9lnl9ospj757rq75kiumttqconhnfcc.apps.googleusercontent.com',
-    webClientId: '1080245081969-u9lnl9ospj757rq75kiumttqconhnfcc.apps.googleusercontent.com',
-  });
+  let handleOnLoginWithGoogle;
 
-  const handleOnLoginWithGoogle = useCallback(async () => {
-    try {
-      const result = await promptAsync();
+  // skips the code during test
+  if (process.env.NODE_ENV !== 'test') {
+    const [request, response, promptAsync] = Google.useAuthRequest({
+      responseType: 'id_token',
+      expoClientId: '1080245081969-u9lnl9ospj757rq75kiumttqconhnfcc.apps.googleusercontent.com',
+      iosClientId: '1080245081969-u9lnl9ospj757rq75kiumttqconhnfcc.apps.googleusercontent.com',
+      androidClientId: '1080245081969-u9lnl9ospj757rq75kiumttqconhnfcc.apps.googleusercontent.com',
+      webClientId: '1080245081969-u9lnl9ospj757rq75kiumttqconhnfcc.apps.googleusercontent.com',
+    });
```

Comment:
Are these supposed to be secret credentials? If they are, they should be changed asap if possible and then put in environment variables for the future since the repo's public.

Reply:
Yeah, we should move these into secrets. I believe I was in a rush to migrate from GoogleOAuth into using ExpoAuth Providers.

Reply:
Ok, I'll try and move them to secrets, and I'll fix the yarn/npm problem soon.